src/ralph/backends/data/mongo.py

```diff
@@ -0,0 +1,399 @@
+from ralph.backends.data.base import (
+    BaseDataBackend,
+    BaseDataBackendSettings,
+    BaseOperationType,
+    BaseQuery,
+    DataBackendStatus,
+    StatementParameters,
+    StatementQueryResult,
+    enforce_query_checks,
+)
+from ralph.conf import BaseSettingsConfig
+from pathlib import Path
+from pydantic import BaseModel, BaseSettings, ValidationError
+
+logger = logging.getLogger(__name__)
+
+class MongoDataBackendSettings(BaseDataBackendSettings):
+    """Represents the Mongo data backend default configuration.
+
+    Attributes:
+        CONNECTION_URI (str): The MongoDB connection URI.
+        DATABASE (str): The MongoDB database to connect to.
+        COLLECTION (str): The MongoDB database collection to get objects from.
+        CLIENT_OPTIONS (MongoClientOptions): A dictionary of valid options
+    """
+
+    class Config(BaseSettingsConfig):
+        """Pydantic Configuration."""
+
+        env_prefix = "RALPH_BACKENDS__DATA__MONGO__"
+
+    CONNECTION_URI: str = None
+    DATABASE: str = None
+    COLLECTION: str = None
+    CLIENT_OPTIONS: MongoClientOptions = MongoClientOptions()
+    DEFAULT_QUERY_STRING: str = "*"
+    DEFAULT_CHUNK_SIZE: int = 500
+    LOCALE_ENCODING: str = "utf8"
+
+class MongoQuery(BaseModel):
+    """Mongo query model."""
+
+    filter: Optional[dict]
+    projection: Optional[dict]
+
+
+class MongoDataBackend(BaseDataBackend):
+    """Mongo database backend."""
+
+    name = "mongo"
+    query_model = MongoQuery
+    default_operation_type = BaseOperationType.CREATE
+    settings_class = MongoDataBackendSettings
+
+    def __init__(self, settings: settings_class = None):
+        """Instantiates the Mongo client.
+
+        Args:
+            connection_uri (str): MongoDB connection URI.
+            database (str): MongoDB database to connect to.
+            collection (str): MongoDB database collection to get objects from.
+            client_options (MongoClientOptions): A dictionary of valid options
+            for the MongoClient class initialization.
+        """
+        self.client = MongoClient(settings.CONNECTION_URI, **settings.CLIENT_OPTIONS.dict())
+        self.database = getattr(self.client, settings.DATABASE)
+        self.collection = getattr(self.database, settings.COLLECTION)
+        self.default_chunk_size = settings.DEFAULT_CHUNK_SIZE
+        self.locale_encoding = settings.LOCALE_ENCODING
+
+    def status(self) -> DataBackendStatus:
+        """Checks MongoDB cluster connection status."""
+        # Check Mongo cluster connection
+        try:
+            self.client.admin.command("ping")
+        except ConnectionFailure:
+            return DataBackendStatus.AWAY
+
+        # Check cluster status
+        if self.client.admin.command("serverStatus").get("ok", 0.0) < 1.0:
+            return DataBackendStatus.ERROR
+
+        return DataBackendStatus.OK
+
+    def list(
+        self, target: str = None, details: bool = False, new: bool = False
```

Comment:
We could be able to change the `collection` we are trying to list documents from, to the `target` parameter, or we will only be able to list to the default collection set in the settings.
That seems to be the case for the read and write method also.